sway-core/tests/sway_to_ir/break.ir

```diff
@@ -0,0 +1,43 @@
+script {
+    fn main() -> (), !1 {
+        entry:
+        br while
+
+        while:
+        v0 = const bool true, !2
+        cbr v0, while_body, end_while
+
+        while_body:
+        br while0
+
+        end_while:
+        v1 = const unit ()
+        ret () v1
```

Comment:
This is out of order and a bit confusing.  I've tried fairly hard to make most jumps go downwards and for `ret` to be at the end of the function.  I still need to review the IRgen so maybe it's a bit hard to predict ahead of time what blocks are needed when there could be many breaks or continues pending... maybe we should reorder the blocks after finalising a function and/or create a mechanism for inserting new blocks at a cursor rather than always appending.

Reply:
Yeah those two options you mentioned are possible! I had to do this unfortunate reordering because I need to know the final block before processing the body. That being said, some future loop optimization passes may decide to reorder those blocks anyways (things like loop rotation) so readability of the IR may be compromised in the future for other reasons. But everything functions well with reordered blocks so that's a good sign!

Reply:
True, readability isn't really number-1 priority.